server/src/main/java/org/elasticsearch/search/aggregations/metrics/TDigestState.java

```diff
@@ -7,33 +7,138 @@
  */
 package org.elasticsearch.search.aggregations.metrics;
 
+import org.elasticsearch.TransportVersion;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.tdigest.AVLTreeDigest;
 import org.elasticsearch.tdigest.Centroid;
+import org.elasticsearch.tdigest.TDigest;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
 import java.util.Iterator;
+import java.util.List;
 
 /**
- * Extension of {@link org.elasticsearch.tdigest.TDigest} with custom serialization.
+ * Decorates {@link org.elasticsearch.tdigest.TDigest} with custom serialization. The underlying implementation for TDigest is selected
+ * through factory method params, providing one optimized for performance (e.g. MergingDigest or HybridDigest) by default, or optionally one
+ * that produces highly accurate results regardless of input size but its construction over the sample population takes 2x-10x longer.
  */
-public class TDigestState extends AVLTreeDigest {
+public class TDigestState {
+
+    /**
+     * Indicates which implementation is used in TDigestState.
+     */
+    public enum ExecutionHint {
```

Comment:
Can we parse to this enum earlier? This way we can already fail on coordinating node.
Maybe instead of using a string in builders/percentileconfig|method use this enum?

Reply:
Done.